hnn_core/parallel_backends.py

```diff
@@ -652,7 +671,8 @@ def __exit__(self, type, value, traceback):
         _BACKEND = self._old_backend
 
         # always kill nrniv processes for good measure
-        kill_proc_name('nrniv')
+        if self.n_procs > 1:
+            kill_proc_name('nrniv')
```

Comment:
Does the `nrniv` process not get created for `n_procs == 1`?

Reply:
Oh I see, this constrains when `_get_procs_running` and `_kill_procs` are called which each require `psutil`. Why do you need to import `psutil` locally within each of those functions?

Also, why not make them methods of `MPIBackend` if they are only used within that class?

Reply:
`self.n_procs = 1` makes MPIBackend default to JoblibBackend. It's a weird way to do it but that's how it was set up and probably makes debugging easier.

The local import of `psutil` makes it a soft dependency. That is, if you do `import hnn_core` you don't get an error if `psutil` is not installed. You get a warning only if trying to use `MPIBackend` and it's not installed. 

As a general rule you don't want to make things methods just to group them together. For that you have files and modules, so you can put them together. If the function requires access to the object attributes, it might make sense. 

Also, I usually like to keep very few things within a class and try to avoid class when possible. Adding everything as attributes can create very spaghetti code.

Reply:
I see, that makes sense.